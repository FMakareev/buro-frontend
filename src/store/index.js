/* global isBrowser */
import { createStore, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';

import { composeWithDevTools } from 'redux-devtools-extension';
import { rootReducer } from './reducers/index';

const createStoreMiddleware = applyMiddleware(thunkMiddleware)(createStore);
const initialState = isBrowser ? window.PRELOADED_REDUX_STATE : {};
initialState.user = isBrowser ? JSON.parse(localStorage.getItem('user')) : null;

const Store = createStoreMiddleware(rootReducer, initialState, composeWithDevTools());

export { Store };

export default Store;
