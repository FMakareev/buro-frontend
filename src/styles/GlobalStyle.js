import { createGlobalStyle } from 'styled-components';
import BG from '../assets/image/bg.jpg';

export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
    font-family: 'Montserrat Regular';
    font-size: 16px;
  }
  #app{
    background-color: #fff;
    background-image: url(${BG});
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
  }
  a {
    text-decoration: none;
    color: inherit;
  }
  
  button:focus, input:focus, select:focus {
    outline: none !important;
    box-shadow: 0 0 0 5px #E7EDF8;
  }
  

  @font-face {
    font-family: 'Montserrat Regular';
    src: url('../assets/fonts/Montserrat-Regular.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
  }

  @font-face {
  font-family: 'Montserrat Medium';
  src: url('../assets/fonts/Montserrat-Medium.ttf') format('truetype');
  font-weight: normal;
  font-style: normal;
  }

  @font-face {
  font-family: 'Montserrat Bold';
  src: url('../assets/fonts/Montserrat-Bold.ttf') format('truetype');
  font-weight: bold;
  font-style: normal;
}

`;

export default GlobalStyle;
