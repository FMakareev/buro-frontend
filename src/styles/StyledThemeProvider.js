import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { TextVariant } from './variants/textVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#093971',
  color2: '#EFF9FC',
  color3: '#E7EDF8',
  color4: '#61C3E2',
  color5: '#A2A2A2',
  color6: '#0F60BD',
  color7: '#072C57',
  color8: '#61C3E2',
  color9: '#EB5757',
  color10: '#E7EDF8',
  color11: '',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  14,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['576px', '768px', '992px', '1200px'];

const boxShadow = ['4px 8px 16px rgba(28, 65, 84, 0.08)'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        text: TextVariant,
      },
      fontFamily: {
        regular: 'Montserrat Regular',
        medium: 'Montserrat Medium',
        bold: 'Montserrat Bold',
      },
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
