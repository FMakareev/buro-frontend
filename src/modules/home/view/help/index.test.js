import React from 'react';
import renderer from 'react-test-renderer';
import { Provider as ProviderRedux } from 'react-redux';
import { HelpPage } from './index';
import { StyledThemeProvider } from '../../../../styles/StyledThemeProvider';
import { Store } from '../../../../store';

test('HelpPage: рендер без ошибок', () => {
  const output = renderer.create(
    <StyledThemeProvider>
      <ProviderRedux store={Store}>
        <HelpPage />
      </ProviderRedux>
    </StyledThemeProvider>,
  );
  expect(output).toMatchSnapshot();
});
