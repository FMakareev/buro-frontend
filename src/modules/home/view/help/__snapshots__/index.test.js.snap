// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpPage: рендер без ошибок 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <div
    className="sc-1jxwge6-0 sc-14k9gvj-0 jjyzlr-0 hjCnhY"
  >
    <div
      className="sc-1jxwge6-0 gHyFgL"
    >
      <div
        className="sc-1jxwge6-0 sc-18ch56c-0 dVmHqH"
        color="color1"
        fontFamily="bold"
        fontSize={9}
      >
        FAQ
      </div>
      <div
        className="sc-1jxwge6-0 jHDLGC"
      >
        <div
          className="y2tw4m-0 gzOhJN"
          onClick={[Function]}
        />
        <div
          className="sc-7zx66l-0 eXlZbJ"
        >
          <div
            className="sc-1jxwge6-0 wterhv-0 sc-185hykj-0 bZsOMn"
            onClick={[Function]}
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 fxqvUH"
              color="color1"
            >
              Unde placeat eos nulla eligendi qui.
            </div>
            <svg
              fill="inherit"
              height="20px"
              viewBox="0 0 32 32"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.94652 14.8535C8.75117 14.6582 8.75117 14.3418 8.94652 14.1465C9.14181 13.9512 9.45821 13.9512 9.65351 14.1465L16.3 20.7929L22.9465 14.1465C23.1418 13.9512 23.4582 13.9512 23.6535 14.1465C23.7511 14.2441 23.8 14.3721 23.8 14.5C23.8 14.6279 23.7511 14.7558 23.6535 14.8535L16.6535 21.8535C16.4582 22.0488 16.1418 22.0488 15.9465 21.8535L8.94652 14.8535Z"
                fill="inherit"
              />
              <path
                d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.271 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                fill="inherit"
              />
            </svg>
          </div>
        </div>
        <div
          className="y2tw4m-0 gzOhJN"
          onClick={[Function]}
        />
        <div
          className="sc-7zx66l-0 eXlZbJ"
        >
          <div
            className="sc-1jxwge6-0 wterhv-0 sc-185hykj-0 bZsOMn"
            onClick={[Function]}
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 fxqvUH"
              color="color1"
            >
              Saepe laboriosam aliquid sit voluptatem vel cum et sapiente.
            </div>
            <svg
              fill="inherit"
              height="20px"
              viewBox="0 0 32 32"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.94652 14.8535C8.75117 14.6582 8.75117 14.3418 8.94652 14.1465C9.14181 13.9512 9.45821 13.9512 9.65351 14.1465L16.3 20.7929L22.9465 14.1465C23.1418 13.9512 23.4582 13.9512 23.6535 14.1465C23.7511 14.2441 23.8 14.3721 23.8 14.5C23.8 14.6279 23.7511 14.7558 23.6535 14.8535L16.6535 21.8535C16.4582 22.0488 16.1418 22.0488 15.9465 21.8535L8.94652 14.8535Z"
                fill="inherit"
              />
              <path
                d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.271 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                fill="inherit"
              />
            </svg>
          </div>
        </div>
        <div
          className="y2tw4m-0 gzOhJN"
          onClick={[Function]}
        />
        <div
          className="sc-7zx66l-0 eXlZbJ"
        >
          <div
            className="sc-1jxwge6-0 wterhv-0 sc-185hykj-0 bZsOMn"
            onClick={[Function]}
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 fxqvUH"
              color="color1"
            >
              Cumque ipsum maxime occaecati.
            </div>
            <svg
              fill="inherit"
              height="20px"
              viewBox="0 0 32 32"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.94652 14.8535C8.75117 14.6582 8.75117 14.3418 8.94652 14.1465C9.14181 13.9512 9.45821 13.9512 9.65351 14.1465L16.3 20.7929L22.9465 14.1465C23.1418 13.9512 23.4582 13.9512 23.6535 14.1465C23.7511 14.2441 23.8 14.3721 23.8 14.5C23.8 14.6279 23.7511 14.7558 23.6535 14.8535L16.6535 21.8535C16.4582 22.0488 16.1418 22.0488 15.9465 21.8535L8.94652 14.8535Z"
                fill="inherit"
              />
              <path
                d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.271 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                fill="inherit"
              />
            </svg>
          </div>
        </div>
        <div
          className="y2tw4m-0 gzOhJN"
          onClick={[Function]}
        />
        <div
          className="sc-7zx66l-0 eXlZbJ"
        >
          <div
            className="sc-1jxwge6-0 wterhv-0 sc-185hykj-0 bZsOMn"
            onClick={[Function]}
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 fxqvUH"
              color="color1"
            >
              Temporibus tempore accusantium.
            </div>
            <svg
              fill="inherit"
              height="20px"
              viewBox="0 0 32 32"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.94652 14.8535C8.75117 14.6582 8.75117 14.3418 8.94652 14.1465C9.14181 13.9512 9.45821 13.9512 9.65351 14.1465L16.3 20.7929L22.9465 14.1465C23.1418 13.9512 23.4582 13.9512 23.6535 14.1465C23.7511 14.2441 23.8 14.3721 23.8 14.5C23.8 14.6279 23.7511 14.7558 23.6535 14.8535L16.6535 21.8535C16.4582 22.0488 16.1418 22.0488 15.9465 21.8535L8.94652 14.8535Z"
                fill="inherit"
              />
              <path
                d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.271 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                fill="inherit"
              />
            </svg>
          </div>
        </div>
        <div
          className="y2tw4m-0 gzOhJN"
          onClick={[Function]}
        />
        <div
          className="sc-7zx66l-0 eXlZbJ"
        >
          <div
            className="sc-1jxwge6-0 wterhv-0 sc-185hykj-0 bZsOMn"
            onClick={[Function]}
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 fxqvUH"
              color="color1"
            >
              Exercitationem aut in vitae expedita cupiditate sunt et.
            </div>
            <svg
              fill="inherit"
              height="20px"
              viewBox="0 0 32 32"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.94652 14.8535C8.75117 14.6582 8.75117 14.3418 8.94652 14.1465C9.14181 13.9512 9.45821 13.9512 9.65351 14.1465L16.3 20.7929L22.9465 14.1465C23.1418 13.9512 23.4582 13.9512 23.6535 14.1465C23.7511 14.2441 23.8 14.3721 23.8 14.5C23.8 14.6279 23.7511 14.7558 23.6535 14.8535L16.6535 21.8535C16.4582 22.0488 16.1418 22.0488 15.9465 21.8535L8.94652 14.8535Z"
                fill="inherit"
              />
              <path
                d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.271 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                fill="inherit"
              />
            </svg>
          </div>
        </div>
        <div
          className="y2tw4m-0 gzOhJN"
          onClick={[Function]}
        />
        <div
          className="sc-7zx66l-0 eXlZbJ"
        >
          <div
            className="sc-1jxwge6-0 wterhv-0 sc-185hykj-0 bZsOMn"
            onClick={[Function]}
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 fxqvUH"
              color="color1"
            >
              Ut voluptatem repellendus et nesciunt officiis in.
            </div>
            <svg
              fill="inherit"
              height="20px"
              viewBox="0 0 32 32"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.94652 14.8535C8.75117 14.6582 8.75117 14.3418 8.94652 14.1465C9.14181 13.9512 9.45821 13.9512 9.65351 14.1465L16.3 20.7929L22.9465 14.1465C23.1418 13.9512 23.4582 13.9512 23.6535 14.1465C23.7511 14.2441 23.8 14.3721 23.8 14.5C23.8 14.6279 23.7511 14.7558 23.6535 14.8535L16.6535 21.8535C16.4582 22.0488 16.1418 22.0488 15.9465 21.8535L8.94652 14.8535Z"
                fill="inherit"
              />
              <path
                d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.271 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                fill="inherit"
              />
            </svg>
          </div>
        </div>
        <div
          className="y2tw4m-0 gzOhJN"
          onClick={[Function]}
        />
        <div
          className="sc-7zx66l-0 eXlZbJ"
        >
          <div
            className="sc-1jxwge6-0 wterhv-0 sc-185hykj-0 bZsOMn"
            onClick={[Function]}
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 fxqvUH"
              color="color1"
            >
              Rerum rerum facilis dolor.
            </div>
            <svg
              fill="inherit"
              height="20px"
              viewBox="0 0 32 32"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.94652 14.8535C8.75117 14.6582 8.75117 14.3418 8.94652 14.1465C9.14181 13.9512 9.45821 13.9512 9.65351 14.1465L16.3 20.7929L22.9465 14.1465C23.1418 13.9512 23.4582 13.9512 23.6535 14.1465C23.7511 14.2441 23.8 14.3721 23.8 14.5C23.8 14.6279 23.7511 14.7558 23.6535 14.8535L16.6535 21.8535C16.4582 22.0488 16.1418 22.0488 15.9465 21.8535L8.94652 14.8535Z"
                fill="inherit"
              />
              <path
                d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.271 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                fill="inherit"
              />
            </svg>
          </div>
        </div>
        <div
          className="y2tw4m-0 gzOhJN"
          onClick={[Function]}
        />
        <div
          className="sc-7zx66l-0 eXlZbJ"
        >
          <div
            className="sc-1jxwge6-0 wterhv-0 sc-185hykj-0 bZsOMn"
            onClick={[Function]}
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 fxqvUH"
              color="color1"
            >
              Omnis aut voluptas qui nesciunt molestias corporis error voluptatum laboriosam.
            </div>
            <svg
              fill="inherit"
              height="20px"
              viewBox="0 0 32 32"
              width="20px"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.94652 14.8535C8.75117 14.6582 8.75117 14.3418 8.94652 14.1465C9.14181 13.9512 9.45821 13.9512 9.65351 14.1465L16.3 20.7929L22.9465 14.1465C23.1418 13.9512 23.4582 13.9512 23.6535 14.1465C23.7511 14.2441 23.8 14.3721 23.8 14.5C23.8 14.6279 23.7511 14.7558 23.6535 14.8535L16.6535 21.8535C16.4582 22.0488 16.1418 22.0488 15.9465 21.8535L8.94652 14.8535Z"
                fill="inherit"
              />
              <path
                d="M16 32C24.8365 32 32 24.8365 32 16C32 7.16345 24.8365 0 16 0C7.1634 0 0 7.16345 0 16C0 24.8366 7.1634 32 16 32ZM16 31C7.72895 31 1 24.271 1 16C1 7.729 7.72895 1 16 1C24.271 1 31 7.729 31 16C31 24.271 24.271 31 16 31Z"
                fill="inherit"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
